### PR TITLE
Validation test

### PR DESCRIPTION
**Pull request by girma336**
## In this project, I added validations to blog post models and create specs for them.

- [ ]  Added updated Gemfile
- [ ]  Added Specs
- [ ] Added validation for model

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      user = User.new(name: 'Tom', posts_counter: -1)
+      expect(user).to_not be_valid
+      expect(user.errors[:posts_counter]).to include('must be greater than or equal to 0')
+    end
+
+    it 'is invalid with a negative posts counter' do
+      user = User.new(name: 'Tom', posts_counter: 'string')
+      expect(user).to_not be_valid
+      expect(user.errors[:posts_counter]).to include('is not a number')
+    end
```
Comment (AshakaE) [inline spec/models/user_spec.rb]: - [ ] Kindly correct your `it` blocks to define what the test checks for.

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      expect(author.errors[:likes_counter]).to include('is not a number')
+    end
+
+    it 'title should ne presence' do
+      user2 = User.new(name: 'girma', posts_counter: 1)
+      user2.save
+      author = Post.new(author: user2, title: 'Hello', likes_counter: 0, comments_counter: -11)
+      expect(author).to_not be_valid
+      expect(author.errors[:comments_counter]).to include('must be greater than or equal to 0')
+    end
```
Comment (AshakaE) [inline spec/models/post_spec.rb]: - [ ] Kindly correct your `it` blocks it hard to understand what your test cases do

File: spec/models/post_spec.rb [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      expect(author).to be_valid
+    end
+
+    it 'title should ne presence' do
+      user2 = User.new(name: 'girma', posts_counter: 1)
+      user2.save
+      author = Post.new(author: user2, title: 'Hello', likes_counter: -80)
+      expect(author).to_not be_valid
+      expect(author.errors[:likes_counter]).to include('must be greater than or equal to 0')
+    end
```
Comment (AshakaE) [inline spec/models/post_spec.rb]: - [ ] Kindly correct your `it` blocks it hard to understand what your test cases do

File: spec/models/post_spec.rb [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    expect(subject.title).to eq('Hello')
+  end
+
+  it 'text should be given strirng' do
+    expect(subject.text).to eq('This is my first post')
+  end
+
+  it 'author shold be ' do
+    expect(subject.author).to eq(first_user)
+  end
```
Comment (AshakaE) [inline spec/models/post_spec.rb]: - [ ] Kindly correct your `it` blocks it hard to understand what your test cases do

**Review by AshakaE — CHANGES_REQUESTED** [review]
Hi @girma336 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Great effort writing the test cases, it still needs some polishing 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/post_spec.rb [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+      user2 = User.new(name: 'girma', posts_counter: 1)
+      user2.save
+      author = Post.new(author: user2, title: 'Hello', likes_counter: -80)
+      expect(author).to_not be_valid
+      expect(author.errors[:likes_counter]).to include('must be greater than or equal to 0')
+    end
+  end
+
+  context 'validation' do
+    it 'title should be presence' do
```
Comment (franclobo) [inline spec/models/post_spec.rb]: - [ ] As requested by the previous code reviewer please 🙏🏽 correct the `it` description to understand what the test does. May you want to say "present"?

> Kindly correct your it blocks it hard to understand what your test cases do

File: spec/models/post_spec.rb [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+  it 'text should be equal to This is my first post' do
+    expect(subject.text).to eq('This is my first post')
+  end
+
+  it 'author shold be equla to first_user' do
+    expect(subject.author).to eq(first_user)
+  end
+
+  context 'Check all Validation' do
+    it 'auther should be presence' do
```
Comment (franclobo) [inline spec/models/post_spec.rb]: - [ ] As requested by the previous code reviewer please 🙏🏽 correct the `it` description to understand what the test does. May you want to say "**author** should be **present**"?

> Kindly correct your it blocks it hard to understand what your test cases do

File: spec/models/post_spec.rb [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+      author = Post.new(title: 'Hello')
+      expect(author).to_not be_valid
+      expect(author.errors[:author]).to include('must exist')
+    end
+
+    it 'text of post shoud be equ to My name ...' do
```
Comment (franclobo) [inline spec/models/post_spec.rb]: - [ ] As requested by the previous code reviewer please 🙏🏽 correct the `it` description to understand what the test does. May you want to say "text of post **should** be **equal** to My name ..."?

> Kindly correct your it blocks it hard to understand what your test cases do

File: spec/models/post_spec.rb [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+      user2 = User.new(name: 'girma', posts_counter: 1)
+      user2.save
+      author = Post.new(author: user2, title: 'Hello', text: 'My name is girma')
+      expect(author.text).to eq('My name is girma')
+    end
+
+    it 'title should be  equla to Hello' do
```
Comment (franclobo) [inline spec/models/post_spec.rb]: - [ ] As requested by the previous code reviewer please 🙏🏽 correct the `it` description to understand what the test does. May you want to say "equal"?

> Kindly correct your it blocks it hard to understand what your test cases do

File: spec/models/post_spec.rb [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+      user2 = User.new(name: 'girma', posts_counter: 1)
+      user2.save
+      author = Post.new(author: user2, title: 'Hello', likes_counter: 0)
+      expect(author.title).to eq('Hello')
+    end
+
+    it 'like counter should be greater than or eql to 0' do
```
Comment (franclobo) [inline spec/models/post_spec.rb]: - [ ] As requested by the previous code reviewer please 🙏🏽 correct the `it` description to understand what the test does. May you want to say "equal"?

> Kindly correct your it blocks it hard to understand what your test cases do

File: spec/models/post_spec.rb [hunk 10]
```diff
@@ -0,0 +1,77 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
+  subject { Post.create(author: first_user, title: 'Hello', text: 'This is my first post') }
+  it 'should be inheritanc of' do
+    expect(Post < ApplicationRecord).to eq(true)
+  end
+  before { subject.save }
+  it 'title should be equle to Hello' do
```
Comment (franclobo) [inline spec/models/post_spec.rb]: - [ ] As requested by the previous code reviewer please 🙏🏽 correct the `it` description to understand what the test does. May you want to say "equal"?

> Kindly correct your it blocks it hard to understand what your test cases do

File: spec/models/post_spec.rb [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+    expect(subject.title).to eq('Hello')
+  end
+
+  it 'text should be equal to This is my first post' do
+    expect(subject.text).to eq('This is my first post')
+  end
+
+  it 'author shold be equla to first_user' do
```
Comment (franclobo) [inline spec/models/post_spec.rb:18]: - [ ] As requested by the previous code reviewer please 🙏🏽 correct the `it` description to understand what the test does. May you want to say "author **should** be **equal** to first_user"?

> Kindly correct your it blocks it hard to understand what your test cases do

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @girma336, 

# ♻️ Required Changes ♻️

soClose

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- No linter errors ✔️ 
- All tests passed 💯 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by algerina — APPROVED** [review]
Hi @girma336  ,

Great job so far! 

May I highlight the following:
- The repo looks professional.:heavy_check_mark:
- Your tests descriptions are meaningful PR:heavy_check_mark:

## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:

Your project is complete!  It's time to merge it :shipit:



 
 ## Optional suggestions

 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)

_Reviewed by [Amel Khiri ](https://github.com/algerina)_

**Pull request merged**